src/v0/destinations/salesforce/utils.js

```diff
@@ -33,16 +33,22 @@ const salesforceResponseHandler = (destResponse, sourceMessage, authKey, authori
     const matchErrorCode = (errorCode) =>
       response && Array.isArray(response) && response.some((resp) => resp?.errorCode === errorCode);
     if (status === 401 && authKey && matchErrorCode('INVALID_SESSION_ID')) {
-      if (authorizationFlow === LEGACY) {
-        // checking for invalid/expired token errors and evicting cache in that case
-        // rudderJobMetadata contains some destination info which is being used to evict the cache
-        ACCESS_TOKEN_CACHE.del(authKey);
+      if (authorizationFlow === OAUTH) {
+        throw new RetryableError(
+          `${DESTINATION} Request Failed - due to "INVALID_SESSION_ID", (Retryable) ${sourceMessage}`,
+          500,
+          destResponse,
+          getAuthErrCategoryFromStCode(status),
+        );
       }
+      // checking for invalid/expired token errors and evicting cache in that case
+      // rudderJobMetadata contains some destination info which is being used to evict the cache
+      ACCESS_TOKEN_CACHE.del(authKey);
       throw new RetryableError(
         `${DESTINATION} Request Failed - due to "INVALID_SESSION_ID", (Retryable) ${sourceMessage}`,
         500,
         destResponse,
-        authorizationFlow === LEGACY ? '' : getAuthErrCategoryFromStCode(status),
+        '',
```

Comment:
No need to pass empty string for this. This field is only needed for Oauth

Reply:
removed